server/app/models/grid_domain_authorization.rb

```diff
@@ -1,14 +1,23 @@
+require 'symmetric-encryption'
+
 class GridDomainAuthorization
   include Mongoid::Document
   include Mongoid::Timestamps
   include Mongoid::Enum
 
   belongs_to :grid
 
-  enum :state, [:created, :requested, :validated], default: :created
+  enum :state, [:created, :requested, :validated, :error], default: :created
+
   field :domain, type: String
   field :challenge, type: Hash
   field :challenge_opts, type: Hash # TODO encrypt?
+  field :authorization_type, type: String, default: 'tls-sni-01'
+  field :service_deploy_id, type: String
```

Comment:
Should this still be fixed to not be a String, aka `belongs_to :grid_service_deploy` before merging? It would be annoying to fix the model later...